edge-modules/iotedge-diagnostics-dotnet/src/GetSocket.cs

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+namespace Diagnostics
+{
+    using System;
+    using System.IO;
+    using System.Net;
+    using System.Net.Sockets;
+    using System.Text;
+
+    public class GetSocket
+    {
+        public static string GetSocketResponse(string server, string endpoint)
+        {
+            string request = $"GET {endpoint} HTTP/1.1\r\nHost: {server}\r\nConnection: Close\r\n\r\n";
+            byte[] bytesSent = Encoding.ASCII.GetBytes(request);
+            byte[] bytesReceived = new byte[256];
+            server = server.Replace("unix://", string.Empty);
```

Comment:
check server is not null or empty.

Reply:
If it is, it throws an exception, which is the intended behavior